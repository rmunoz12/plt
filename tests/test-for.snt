@setup {

group A(Object) {
    int x;
    int y;
    func group A __init__(int x, int y) {
        this.x = x;
        this.y = y;
        return this;
    }

    func int test() {
        return this.x + this.y;
    }
};

}

@turns {

func void begin() {
    for (int x in {1, 2, 3}) {
        print(x); print("\n");
    }
    pass(test_str, 0);
}

func void test_str() {
    for (str s in {"abc", "def", "ghi", "jkl"}) {
        print(s); print("\n");
    }
    pass(test_grp, 0);
}

func void test_grp() {
    group A a1 = A(1, 2);
    group A a2 = A(3, 4);
    group A a3 = A(5, 6);
    for (group A x in {a1, a2, a3}) {
        print(x.y); print(" ");
        print(x.test()); print("\n");
    }
    pass(test_list, 0);
}

func void test_list() {
    list[int] a = [1, 2, 3];
    list[int] b = [4, 5];
    list[int] c = [6, 7];
    int y;
    for (list[int] x in {a, b, c}) {
        y = x[0];
        print(y); print("\n");
        print(x); print("\n");
    }
    end;
}

}
