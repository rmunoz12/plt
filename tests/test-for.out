1
2
3
abc
def
ghi
jkl
2 3
4 7
6 11
1
[1, 2, 3]
4
[4, 5]
6
[6, 7]
